### PR TITLE
Add com.github.ahrm.sioyek pdf viewer

### PR DESCRIPTION
**Pull request by nbenitez**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/ahrm/sioyek/issues/163#issuecomment-1059733271
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by nbenitez** [comment]
bot, build com.github.ahrm.sioyek

File: org.boost.Boost.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        "./bootstrap.sh",
+        "./b2 install --prefix=/app --with-system --with-random define=\"BOOST_SYSTEM_NO_DEPRECATED\" -sNO_BZIP2=1 cxxflags=-fPIC cflags=-fPIC"
+    ],
+    "sources":[
+    {
+        "type": "archive",
+        "url": "https://boostorg.jfrog.io/artifactory/main/release/1.73.0/source/boost_1_73_0.tar.bz2",
+        "sha256": "4eb3b8d442b426dc35346235c8733b5ae35ba431690e38c6a8263dce9fcbb402",
+        "#url": "https://dl.bintray.com/boostorg/release/1.65.1/source/boost_1_65_1.tar.gz",
+        "#sha256": "a13de2c8fbad635e6ba9c8f8714a0e6b4264b60a29b964b940a22554705b6b60"
```
Comment (hfiguiere) [inline org.boost.Boost.json:14]:
```suggestion
```

File: org.boost.Boost.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+    "name": "boost",
```
Comment (hfiguiere) [inline org.boost.Boost.json:2]: this should be in the main manifest.

File: com.github.ahrm.sioyek.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  - --filesystem=xdg-music
+  - --filesystem=/run/media
+  - --filesystem=/media
+  - --filesystem=/tmp
+  - --share=ipc
+  # Sioyek fails with wayland (invisible window)
+  #- --socket=wayland
+  #- --socket=fallback-x11
+  - --socket=x11
+  - --device=shm
```
[the remark this answers is not in the document]
Comment (nbenitez) [reply com.github.ahrm.sioyek.yaml:22]: Yes it is, I had to add it. Sioyek uses shared-memory for single instance, without `--device=shm` sioyek launches once but fails when you launch it a second time.

File: com.github.ahrm.sioyek.yaml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      - type: patch
+        path: patches/sioyek_build_fix.patch
+      - type: file
+        url: https://github.com/ahrm/sioyek/raw/v1.1.0/resources/sioyek-icon-linux.png
+        sha256: ae06a0a8b08d61ad41ce53bc0ad9813f79a9c111bcc35d011f81b2b08a0067ee
+        dest-filename: sioyek.png
+      - type: file
+        path: com.github.ahrm.sioyek.metainfo.xml
+      - type: file
+        path: com.github.ahrm.sioyek.desktop
```
Comment (hfiguiere) [inline com.github.ahrm.sioyek.yaml:72]: upstream has a desktop file. You should use it instead.

File: com.github.ahrm.sioyek.yaml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      - install -Dm644 -t /app/share/metainfo com.github.ahrm.sioyek.metainfo.xml
+    sources:
+      - type: git
+        url: git://github.com/ahrm/sioyek.git
+        commit: 5c325e7bb340aeb55ec4a4b5678b6cf91bc0ac97
+        tag: 'v1.1.0'
+      - type: patch
+        path: patches/sioyek_build_fix.patch
+      - type: file
+        url: https://github.com/ahrm/sioyek/raw/v1.1.0/resources/sioyek-icon-linux.png
```
Comment (hfiguiere) [inline com.github.ahrm.sioyek.yaml:66]: you don't need this. Just take it from the tree checkout above.

File: com.github.ahrm.sioyek.yaml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+      - /include
+      - /lib/*.a
+      - /lib/*.la
+      - /lib/*.prl
+      - /lib/cmake
+      - /lib/mkspecs
+      - /lib/pkgconfig
+    sources:
+      - type: archive
+        url: http://qt.mirror.constant.com/archive/qt/5.12/5.12.12/submodules/qtwayland-everywhere-src-5.12.12.tar.xz
```
[the remark this answers is not in the document]
Comment (nbenitez) [reply com.github.ahrm.sioyek.yaml:41]: This app is not using any KDE api at all, nor any GNOME api, so I prefer to base in Freedesktop runtime so it's better wrt download size and dependency wise.

[the remark this answers is not in the document]
Comment (hfiguiere) [reply com.github.ahrm.sioyek.yaml:41]: no. You should use the KDE runtime.

File: com.github.ahrm.sioyek.yaml (hunk 5 above)
Comment (nbenitez) [reply com.github.ahrm.sioyek.yaml:66]: Done.

File: org.boost.Boost.json (hunk 2 above)
Comment (nbenitez) [reply org.boost.Boost.json:2]: Done.

File: org.boost.Boost.json (hunk 1 above)
Comment (nbenitez) [reply org.boost.Boost.json:14]: Done.

File: com.github.ahrm.sioyek.yaml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+      - ./build_linux.sh
+      - mv build /app/sioyek
+      - ln -s /app/sioyek/sioyek /app/bin/sioyek
+    post-install:
+      - install -Dm644 sioyek.png /app/share/icons/hicolor/256x256/apps/com.github.ahrm.sioyek.png
+      - desktop-file-install --delete-original --dir=/app/share/applications com.github.ahrm.sioyek.desktop
+      - install -Dm644 -t /app/share/metainfo com.github.ahrm.sioyek.metainfo.xml
+    sources:
+      - type: git
+        url: git://github.com/ahrm/sioyek.git
```
[the remark this answers is not in the document]
Comment (nbenitez) [reply com.github.ahrm.sioyek.yaml:60]: Done.

File: com.github.ahrm.sioyek.yaml (hunk 4 above)
Comment (nbenitez) [reply com.github.ahrm.sioyek.yaml:72]: Well that desktop file it's very minimum, that's why I didn't use it, but I will submit patch upstream to add the needed keywords, etc.

**Comment by nbenitez** [comment]
bot, build com.github.ahrm.sioyek

File: com.github.ahrm.sioyek.yaml [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+  - --share=ipc
+  # Sioyek fails with wayland (invisible window)
+  #- --socket=wayland
+  #- --socket=fallback-x11
+  - --socket=x11
+  - --device=shm
+  - --device=dri
+
+modules:
+  - org.boost.Boost.json
```
Comment (hfiguiere) [inline com.github.ahrm.sioyek.yaml:26]: you didn't add the content of the manifest here.

File: com.github.ahrm.sioyek.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (hfiguiere) [reply com.github.ahrm.sioyek.yaml:41]: This remain. There is no "saving" here. The number of patches you carry just to build Qt should already be a good reason to not.

Qt app -> KDE runtime.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by nbenitez** [comment]
bot, build com.github.ahrm.sioyek

File: com.github.ahrm.sioyek.yaml (hunk 8 above)
Comment (nbenitez) [reply com.github.ahrm.sioyek.yaml]: Hopefully now fixed.

File: com.github.ahrm.sioyek.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (barthalion) [reply com.github.ahrm.sioyek.yaml:49]: KDE runtime is for both Qt and KDE apps. Maybe it's not ideal, but maintenance toll is real and it's better to have a single well-maintained runtime for both KDE and Qt than multiple unmaintained ones.

Is there any reason you're using Qt 5.12 here?

[the remark this answers is not in the document]
Comment (nbenitez) [reply com.github.ahrm.sioyek.yaml:49]: Sioyek just needs Qt5. Qt 5.12 is the one I could (after many work) succesfully build and install with Freedesktop runtime 21.08. Qt 5.12 is more than enough for Sioyek (latest is 5.15) and having the latest version is not so important in a containerized system like flatpak.

If Sioyek (or any other app I've submitted with Qt) start depending in a feature or api not covered by Qt 5.12 I will myself update it to latest Qt or to latest KDE runtime.

I understand there's maintance toll in general, but I look after the apps I submit, responding to issues and keeping them working.

[the remark this answers is not in the document]
Comment (nbenitez) [reply com.github.ahrm.sioyek.yaml:49]: Thanks Bartlomiej

**Pull request closed without merging**